### PR TITLE
Add ticket status CRUD.

### PR DESCRIPTION
**Pull request by Bogomazz**

### CODE REVIEW

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        private readonly ITicketStatusService _ticketStatusService;
+
+        public TicketStatusesController(ITicketStatusService ticketStatusService)
+        {
+            _ticketStatusService = ticketStatusService;
+        }
+        
+        [HttpGet()]
+        public async Task<List<TicketStatus>> Get()
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: it is not very good to return domain entity to client. We usually use ResponseModels for that (you can easily map it with automapper)

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        
+        [HttpGet("{id}")]
+        public async Task<TicketStatus> Get(long id)
+        {
+            return await _ticketStatusService.FindById(id);
+        }
+        
+        [HttpPost()]
+        public async Task<TicketStatus> Create(TicketStatus ticketStatus)
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: the same with RequestModels

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        }
+
+        [HttpPut("{id}")]
+        public async Task<TicketStatus> Update(long id, TicketStatus ticketStatus)
+        {
+            return await _ticketStatusService.UpdateAsync(id, ticketStatus);
+        }
+        
+        [HttpDelete("{id}")]
+        public async Task Delete(long id)
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: in such cases it is more preferable not to make method async, but simply return Task, e.g. return _ticketStatusService.DeleteAsync(id)

File: HdProduction.HelpDesk.Infrastructure/Repositories/TicketStatusRepository.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        public void Remove(TicketStatus entity)
+        {
+            _context.TicketStatuses.Remove(entity);
+        }
+
+        public async Task SaveAsync()
+        {
+            await _context.SaveChangesAsync();
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/Repositories/TicketStatusRepository.cs]: return Task:)

File: HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        public async Task<TicketStatus> CreateAsync(TicketStatus ticketStatus)
+        {
+            _ticketStatusRepository.Add(ticketStatus);
+            await _ticketStatusRepository.SaveAsync();
+            return ticketStatus;
+        }
+
+        public async Task<TicketStatus> UpdateAsync(long id, TicketStatus ticketStatus)
+        {
+            var statusEntity = await _ticketStatusRepository.FindAsync(id);
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs]: why don't you check that entity exists?

File: HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        private readonly ITicketStatusRepository _ticketStatusRepository;
+
+        public TicketStatusService(ITicketStatusRepository ticketStatusRepository)
+        {
+            _ticketStatusRepository = ticketStatusRepository;
+        }
+        
+        public async Task<TicketStatus> CreateAsync(TicketStatus ticketStatus)
+        {
+            _ticketStatusRepository.Add(ticketStatus);
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs:20]: as far as I understand you decided to leave validations for better times?

File: HdProduction.HelpDesk.Infrastructure/EfConfigurations/TicketStatusConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace HdProduction.HelpDesk.Infrastructure.EfConfigurations
+{
+    public class TicketStatusConfiguration : EntityBaseConfiguration<TicketStatus, long>
+    {
+        protected override void ConfigureNext(EntityTypeBuilder<TicketStatus> builder)
+        {
+            builder.Property(t => t.Name)
+                .IsRequired()
+                .HasMaxLength(256);
```
Comment (farexy) [inline HdProduction.HelpDesk.Infrastructure/EfConfigurations/TicketStatusConfiguration.cs]: I think 256 is too much for status name. 32 would be better, or even 16..

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs (hunk 1 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: Agree with that. And I want to suggest a slightly different folder structure. Stay in touch.

File: HdProduction.HelpDesk.Infrastructure/EfConfigurations/TicketStatusConfiguration.cs (hunk 7 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Infrastructure/EfConfigurations/TicketStatusConfiguration.cs]: Ok!

File: HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs (hunk 6 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs:20]: Doesn't EF perform model validation?

File: HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs (hunk 5 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs]: My fault.

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs (hunk 1 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: Let's discuss the folder structure in our next meeting. For now, I left it as is.

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs (hunk 2 above)
Comment (Bogomazz) [reply HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: 👍🏻

File: HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs (hunk 6 above)
Comment (farexy) [reply HdProduction.HelpDesk.Infrastructure/Services/TicketStatusService.cs:20]: on database level - yes. but it won't have human-readable  error

File: HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs (hunk 1 above)
Comment (farexy) [reply HdProduction.HelpDesk.Api/Controllers/TicketStatusesController.cs]: ok

File: HdProduction.HelpDesk.Api/Configuration/ErrorHandlingMiddleware.cs [hunk 8]
```diff
@@ -20,16 +21,21 @@ protected override Task HandleExceptionAsync(HttpContext context, Exception exce
...
-      else
-      {
-        Logger.Error("Unknown exception", exception);
-        message = "Oops. Something went wrong.";
+        // TODO normal filters
+        case EntityNotFoundException _:
+          statusCode = HttpStatusCode.NotFound;
+          message = exception.Message;
+          break;
+        case ApplicationException _:
```
Comment (farexy) [inline HdProduction.HelpDesk.Api/Configuration/ErrorHandlingMiddleware.cs]: ApplicationException  was a temp thing, for bad request we usually throw BusinessLogicException (not sure that I've added it)

**Pull request merged**